### PR TITLE
[KYUUBI #5492][AUTHZ] saveAsTable create DataSource table miss db info

### LINKED ISSUE #5492: [Bug] saveAsTable create DataSource table miss db and catalog
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

No db information
截屏2023-10-21 下午3 35 57

### Affects Version(s)

master, 1.8.0

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### _Why are the changes needed?_
To fix #5492 
When we use saveAsTable and write as a DataSource table, since CreateTableAsDataSource command's catalogTable was directly constructed by identifier and only will check when executing, so here authz will miss db information as below case
image

This fixes this issue by following the Spark's code
截屏2023-10-21 下午3 37 37

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_
No

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @bowenliang123 @yaooqinn Could you take a look

**Review by bowenliang123 — APPROVED** [review]
LGTM.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -835,4 +835,19 @@ class HiveCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
...
+    val table1 = "table1"
+    withSingleCallEnabled {
+      withCleanTmpResources(Seq.empty) {
+        val df = doAs(
+          admin,
+          sql(s"SELECT * FROM VALUES(1, 100),(2, 200),(3, 300) AS t(id, scope)")).persist()
+        val e = intercept[AccessControlException] {
+          doAs(someone, df.write.mode("overwrite").saveAsTable(table1))
+        }
+        assert(e.getMessage.contains(s"does not have [create] privilege on [$defaultDb/$table1]"))
```
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:849]: Hi, you may use the `interceptContains` method to simplify the assertions. Very handy and clear.

Comment (AngersZhuuuu) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala]: > Hi, you may use the `interceptContains` method to simplify the assertions. Very handy and clear.

Done. I will refine these test's UT by `interceptContains ` later

**Review by yaooqinn — CHANGES_REQUESTED** [review]
setCurrentDatabaseIfMissing shall be used

**Comment by AngersZhuuuu** [comment]
> setCurrentDatabaseIfMissing shall be used

Done

**Review by cxzl25 — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thanks, merged to master and 1.8

**Comment by yaooqinn** [comment]
Hi @AngersZhuuuu, it conflicts 1.8, can you send a backport PR?

**Pull request closed without merging**